### PR TITLE
feat: added linting config for typescript files

### PR DESCRIPTION
**Pull request by TarunAitharaju**
This PR would add the following changes

1.  Added the `.prettierrc` file
2.  Replaced `google` eslint config with `airbnb` config
3.  ESLint to parse the typescript files

## Summary by CodeRabbit

- **Chores**
  - Updated ESLint configuration to integrate TypeScript support and enhance consistency.
  - Introduced a Prettier configuration file for standardized code formatting.
  - Updated various dependencies and plugins in `package.json` to support the new linting and formatting configurations.
  - Modified the `lint` script to target files in the `src/` directory specifically.

### CODE REVIEW

File: src/examples/FloatingUI/useInteractions/UseInteractionsExample.js [hunk 1]
```diff
@@ -1,63 +1,71 @@
...
+} from "@floating-ui/react";
 
-const UserInteractionsExample = () => {
-    const [isOpen, setIsOpen] = useState(false);
-    const {context, refs, floatingStyles} = useFloating({
-        open: isOpen,
-        placement: 'top',
-        onOpenChange: setIsOpen,
-    });
+function UserInteractionsExample() {
```
Comment (kotAPI) [inline src/examples/FloatingUI/useInteractions/UseInteractionsExample.js:13]: its better we follow es6 syntax, this lint config might not work for us

Comment (TarunAitharaju) [reply src/examples/FloatingUI/useInteractions/UseInteractionsExample.js:13]: That's our next to do item. There are lot of lint errors and warnings that we need to fix. Probably we need a ticket to address all these lint errors and warning

File: package.json [hunk 2]
```diff
@@ -79,24 +79,29 @@
...
+    "eslint-config-airbnb": "19.0.4",
+    "eslint-config-airbnb-typescript": "17.1.0",
+    "eslint-config-airbnb-typescript-prettier": "5.0.0",
+    "eslint-config-prettier": "9.0.0",
+    "eslint-import-resolver-typescript": "^3.6.1",
+    "eslint-plugin-import": "2.28.1",
+    "eslint-plugin-jsx-a11y": "^6.9.0",
+    "eslint-plugin-jest": "27.4.0",
+    "eslint-plugin-jest-dom": "5.1.0",
+    "eslint-plugin-react": "7.33.2",
```
Comment (kotAPI) [inline package.json:96]: this is downgrading

File: .eslintrc.cjs [hunk 3]
```diff
@@ -1,63 +1,81 @@
...
+    "no-undef": "off",
+    "react/no-unescaped-entities": 1,
+    "no-unexpected-multiline": "warn",
+    "no-var": "warn",
+    "prefer-const": "warn",
+    "space-before-blocks": "warn",
+    "space-before-function-paren": ["warn", "never"],
+    "react/jsx-first-prop-new-line": [1, "multiline"],
+    "space-in-parens": "warn",
+    "spaced-comment": "warn",
```
Comment (kotAPI) [inline .eslintrc.cjs:55]: weird, the linting removed single quotes everywhere else, but its adding double quotes here

Comment (TarunAitharaju) [reply .eslintrc.cjs:55]: may be not in the config files..

**Review by kotAPI — APPROVED** [review]

**Pull request merged**